base/setup/lib/utils/partlist.c

```diff
@@ -835,6 +825,144 @@ InitializePartitionEntry(
     return TRUE;
 }
 
+static
+VOID
+InitPartitionDeviceName(
+    _Inout_ PPARTENTRY PartEntry)
+{
+    NTSTATUS Status;
+
+    /* Ignore if this is a container partition */
+    if (IsContainerPartition(PartEntry->PartitionType))
```

Comment:
Would an error message be in order here?

Reply:
No, because this is an expected situation.
In `WritePartitions()` (much below in that file), I call in two places `InitPartitionDeviceName`:
```c
if (PartEntry->New)
{
    PartitionInfo = &DiskEntry->LayoutBuffer->PartitionEntry[PartEntry->PartitionIndex];
    PartEntry->PartitionNumber = PartitionInfo->PartitionNumber;
    InitPartitionDeviceName(PartEntry);
}
PartEntry->New = FALSE;
```
and to simplify the code I don't do the check for container in this block, but instead chose to move it into this private helper function: it can be called in any situation on a partitioned entry, but if this is a container partition, nothing is done, otherwise a device name is generated.